src/test/components/ProfileCallTreeContextMenu.test.js

```diff
@@ -0,0 +1,157 @@
+    const {
+      profile,
+      funcNamesPerThread: [funcNames],
+    } = getProfileFromTextSamples(`
+      A         A         A
+      B:library B:library B:library
+      B:library B:library B:library
+      B:library B:library B:library
+      C         C         H
+      D         F         I
+      E         E
+    `);
+    const store = storeWithProfile(profile);
+    store.dispatch(
+      changeSelectedCallNode(0, [
+        funcNames.indexOf('A'),
+        funcNames.indexOf('B:library'),
+      ])
+    );
+    return store;
+  }
+
+  function setup(store = createStore()) {
+    const wrapper = mount(
+      <Provider store={store}>
+        <ProfileCallTreeContextMenu />
+      </Provider>
+    );
+
+    function findMenuItem(type: string) {
+      return wrapper.find(MenuItem).findWhere(n => {
+        const { data } = n.props();
+        return typeof data === 'object' && data.type === type;
+      });
+    }
+
+    return { store, wrapper, findMenuItem };
+  }
+
+  describe('basic rendering', function() {
+    it('renders a context menu', () => {
+      const { wrapper } = setup();
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+
+  describe('clicking on call tree transforms', function() {
+    // Iterate through each transform slug, and click things in it.
+    for (const type of [
+      'merge-call-node',
+      'merge-function',
+      'focus-subtree',
+      'focus-function',
+      'collapse-function-subtree',
+      'collapse-resource',
+      'collapse-direct-recursion',
+      'drop-function',
+    ]) {
+      it(`adds a transform for "${type}"`, function() {
+        const { store: { getState }, findMenuItem } = setup();
+        findMenuItem(type).simulate('click');
+        expect(
+          selectedThreadSelectors.getTransformStack(getState())[0].type
+        ).toBe(type);
+      });
+    }
+  });
+
+  describe('clicking on the rest of the menu items', function() {
+    it('can expand all call nodes in the call tree', function() {
+      const { store: { getState }, findMenuItem } = setup();
+      expect(
+        selectedThreadSelectors.getExpandedCallNodeIndexes(getState())
+      ).toHaveLength(1);
+
+      findMenuItem('expand-all').simulate('click');
+
+      // This test only asserts that a bunch of call nodes were actually expanded.
+      expect(
+        selectedThreadSelectors.getExpandedCallNodeIndexes(getState())
+      ).toHaveLength(11);
+    });
+
+    it('can look up functions on SearchFox', function() {
+      const { findMenuItem } = setup();
+      jest.spyOn(window, 'open').mockImplementation(() => {});
+      findMenuItem('searchfox').simulate('click');
+      expect(window.open).toBeCalledWith(
+        'https://searchfox.org/mozilla-central/search?q=B%3Alibrary',
+        '_blank'
+      );
+    });
+
+    it('can copy a function name', function() {
+      const { findMenuItem } = setup();
+      // Copy is a mocked module, clear it both before and after.
+      copy.mockClear();
```

Comment:
This mock does not use `jest.spyOn` so it does not auto-clear.

Reply:
Can you look whether adding [resetMocks](https://facebook.github.io/jest/docs/en/configuration.html#resetmocks-boolean) to the configuration (in addition to `restoreMocks` we already have) works, instead of relying on adding `mockClear` at the right places?